src/gt4py/next/iterator/transforms/fuse_as_fieldop.py

```diff
@@ -95,6 +84,41 @@ def _inline_as_fieldop_arg(
     ), extracted_args
 
 
+def _deduplicate_as_fieldop_args(
+    args: dict[str, itir.Expr], stencil_body: itir.Expr
+) -> tuple[dict[str, itir.Expr], itir.Expr]:
+    new_args_inverted: dict[itir.Expr, list[str]] = collections.defaultdict(list)
+    for name, arg in args.items():
+        new_args_inverted[arg].append(name)
+
+    new_args: dict[str, itir.Expr] = {}
+    new_stencil_body = stencil_body
+    for arg, names in new_args_inverted.items():
+        # put internal names at the end
+        names = sorted(names, key=lambda s: (s.startswith("_"), s))
+        new_args[names[0]] = arg
+        if len(names) > 1:
+            new_stencil_body = im.let(*((name, names[0]) for name in names[1:]))(new_stencil_body)
```

Comment:
I would unpack directly  on creation the head and tail of the sorted list (with better names if possible) 
```suggestion
        head, *tail = sorted(names, key=lambda s: (s.startswith("_"), s))
        new_args[head] = arg
        if tail:
            new_stencil_body = im.let(*((name, head) for name in tail))(new_stencil_body)
```

Reply:
Oh, absolutely.